src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+    let initialIndex = 0;
+
+    if (arguments.length < 2 && this.length > 1) {
+      accumulator = this[0];
+      initialIndex++;
```

Comment:
```suggestion
      initialIndex = 1;
```

i think this way is more obvious